core/broadcast.go

```diff
@@ -150,7 +151,14 @@ func (b *broadcast) sendout(h []byte, p packet) {
 	proto := &drand.DKGPacket{
 		Dkg: dkgproto,
 	}
-	b.dispatcher.broadcast(proto)
+	if bypass {
+		// in a routine cause we don't want to block the processing of the DKG
+		// as well - that's ok since we are only expecting to send 3 packets out
+		// at the very least.
+		go b.dispatcher.broadcastDirect(proto)
```

Comment:
having these off in go-routines seems potentially like it could end up with some weird memory leak / build-up of pending writes. I'd prefer a channel and single sending thread for these as well in parallel to the non-bypassing ones.

Reply:
I'm not saying I dont want to do it but I want to highlight that DKG only outputs 1 deal packet, 1 resp packet and up to 1 justification packet.

Reply:
my worry is cases where the sending blocks - e.g. because dns resolution has decided on a bad time to time out for a minute.
In the mean-time, everything can get torn down and a new reshare could start, and then some random packet from this previous thread could still get sent out.
Having it in the same place makes it easier to trust it's going to get torn down properly and not get left dangling.

Reply:
Mhhh but in this specific failure scenario , this can already happen with the current code right ?
I think what we need to prevent against that is using context, so when we stop the broadcaster, we stop all context. Does that seem right ?
And if so, can I or you do that in a subsequent PR ?